include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -624,6 +624,32 @@ typedef struct OrtMIGraphXProviderOptions {
   bool migraphx_exhaustive_tune;                     // migraphx tuned compile  Default = false
 } OrtMIGraphXProviderOptions;
 
+/** \brief WebGPU Execution Provider Options
+ *
+ * When a user wants to use WebGPU as the execution provider, there are 2 ways to specify the WebGPU device:
+ *
+ * 1. Use the default WebGPU device. The default WebGPU device is managed by WebGPU EP internally. The user doesn't
+ *    need to provide any device information in this case. All the fields should be set to nullptr or 0.
+ *
+ * 2. Use a custom WebGPU device. The user should create their own handles of `WGPUInstance`, `WGPUAdapter`, and
+ *    `WGPUDevice` and use arbitrary number in [1..65536) as the device id. The user should provide the handles
+ *    and the device id in the options.
+ *
+ *    When specifying an existing Device ID, the user should provide the handles of `WGPUInstance`, `WGPUAdapter`, and
+ *    `WGPUDevice` in the options. The device id should be the same as the one used previously.
+ *
+ *    It's user's responsibility to manage the lifecycle of the handles and ensure the handles are valid during the
+ *    lifetime of the inference session.
+ *
+ * \see OrtApi::SessionOptionsAppendExecutionProvider_WebGPU
+ */
+typedef struct OrtWebGPUProviderOptions {
+  int device_id;          // WebGPU device id.
```

Comment:
What's this device_id for? Does it only mean the device (and adapter, instance) is not owned by ORT?

Reply:
It's just a number used as key. when the key is the same, it represents the same set of [WGPUInstance, WGPUAdapter, WGPUDevice], and all resources based on the set, including buffer cache manager, program manager and so on.

a better name suggestion is welcome